aws-s3-transfer-manager/src/operation/download.rs

```diff
@@ -129,18 +129,30 @@ fn handle_discovery_chunk(
 
 /// Download operation specific state
 #[derive(Debug)]
-pub(crate) struct DownloadState {}
+pub(crate) struct DownloadState {
+    current_seq: u64,
```

Comment:
Can you not get away with an atomic here?

Reply:
Thanks, updated to AtomicU64.